kafkacat.c

```diff
@@ -307,6 +307,12 @@ static void producer_run (FILE *fp, char **paths, int pathcnt) {
         conf.rkt_conf = NULL;
 
 
+        /* Poll once before initiating any connections.
+         * This is necessary for the SASL/OAUTHBEARER case
+         * to retrieve the initial token but is a no-op otherwise.
```

Comment:
With my modifications to your librdkafka PR, I dont think this is needed anymore since the default unsec JWS token refresh callback now is called automatically from rd_kafka_new().